react-catalog/my-app/src/Catalog.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { readCatalog } from './read';
+import { type Product } from './data';
+import { Link } from 'react-router-dom';
+
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>();
+  useEffect(() => {
+    async function productCall() {
+      try {
+        const catalog = await readCatalog();
+        setProducts(catalog);
+      } catch (error) {
+        console.error(error);
```

Comment:
You need to let the user know that an error occurred. You can use `alert()` or put it somewhere on the page, but the user needs to see it!

How many sites have you used where you click a button and nothing happens? This is usually because the programmer was lazy and did not bother to display the error. Nothing happened because of the error, but you have no way of knowing that. You just think the site is broken. So, _always_ display errors to the user!